docs/operators/s3outputmodule.md

```diff
@@ -0,0 +1,199 @@
+
+### Overview
+------------
+
+We can upload files into S3 through partitioning the AbstractFileOutputOperator. But, S3OutputModule is needed to parallel uploading the blocks of a file into S3. 
+
+Below table illustrates the benefits of this module over AbstractFileOutputOperator.
+
+**S3OutputModule**|**AbstractFileOutputOperator**
+-----|-----
+Maximum upload file size is 5TB.|Maximum upload file size is 5GB.
+Best fit for both large and small files.|Best fit for small files.
+Module uses AmazonS3Client API's to upload objects into S3. Large files will upload using multipart feature and small files(Single Block) will upload using putObject(...) API|Operator uses Hadoop filesystems like S3AFileSystem.Consists of couple of steps to upload object into S3: (1) Write the data into the local filesystem. (2) When the stream closes, filesystem uploads the local object into S3.
+If a block fails to upload then you need to re-upload the data for that block only|If a file fails to upload then you need to re-upload the complete file.
+
+### About Multipart Upload Feature
+----------------------------------
+Uploading the blocks of a file into S3 can be achieved by [multipart feature](http://docs.aws.amazon.com/AmazonS3/latest/dev/mpuoverview.html).
+
+Using Multipart Upload feature, we can upload a single large file in multiple smaller parts. In Multipart Upload process, each part is a portion of object's data. Each part can be uploaded independently. 
+After all parts of a file are uploaded successfully, Amazon S3 combines the parts as a single object.
+
+Please refer the [Java code](http://docs.aws.amazon.com/AmazonS3/latest/dev/llJavaUploadFile.html) for uploading file into Amazon S3 bucket using multipart feature.
+
+### About S3 Output Module
+--------------------------
+
+Purpose of S3Output module is to upload files/directories into an Amazon S3 bucket using the multipart upload feature.
+
+S3Output module is **fault-tolerant**, **statically/dynamically partitionable** and has **exactly once** semantics.
+
+Module class is **S3OutputModule** located in the package **org.apache.apex.malhar.lib.fs.s3**; please refer to [github URL](https://github.com/apache/apex-malhar/blob/master/library/src/main/java/org/apache/apex/malhar/lib/fs/s3/S3OutputModule.java).
+
+### About the operators in S3OutputModule
+-----------------------------------------
+
+Following diagram illustrates the DAG in this module:
+
+![](images/s3output/s3outputmodule.png)
+
+-   ***S3InitateFileUploadOperator***
+    -   Initiate the upload for the file using **AmazonS3Client.initiateMultipartUpload(...)** method only if the number of blocks for a file is greater than 1. By Successfully initiating the file, S3 returns a response of type **InitiateMultipartUploadResult**, which includes the **upload ID**, which is the unique identifier for the multipart upload. This **upload ID** must include whenever upload the blocks, complete an upload of a file.
```

Comment:
By Successfully initiating the file => By successfully initiating the upload

This **upload ID** must include whenever upload the blocks, complete an upload of a file. =>
{Not sure what the intent of the last phrase is, but seems like the following should be adequate}
This **upload ID** must be included whenever additional blocks of the file are uploaded.